txt2hpo/extract.py

```diff
@@ -11,7 +11,232 @@
 from txt2hpo.build_tree import search_tree, build_search_tree
 from txt2hpo.util import remove_key
 
-context_model = load_model()
+
+class Data(object):
+    def __init__(self, entries=None, model=None):
+        if not entries:
+            self.entries = []
+        else:
+            self.entries = entries
+        self.model = model
+
+    def add(self,entry):
+        self.entries += entry
+
+    def resolve_conflicts(self):
+        """
+        Pick most likely HPO ID based on context
+        :param extracted_terms: list of dictionaries identified by extract_hpo_terms
+        :return: list of dictionaries with resolved conflicts
+        """
+
+        if not self.model:
+            logger.critical("Doc2vec model does not exist or could not be loaded")
+
+        resolved_terms = []
+        for entry in self.entries:
+            similarity_scores = []
+            if len(entry['hpid']) > 1:
+                for term in entry['hpid']:
+                    similarity_scores.append(similarity_term_to_context(term, entry['context'], self.model))
+
+                # reduce matches until only one term left
+                for i in range(len(similarity_scores) - 1):
+                    idx_least_likely_term = similarity_scores.index(min(similarity_scores))
+                    least_likely_term = entry['hpid'][idx_least_likely_term]
+                    entry['hpid'].remove(least_likely_term)
+
+            resolved_terms.append(entry)
+        self.entries = resolved_terms
+
+    @property
+    def hpids(self):
+        return [x['hpid'] for x in self.entries]
+
+    @property
+    def json(self):
```

Comment:
I'd suggest renaming this function, `json` is standard library and you might run into namespace collisions.

Reply:
requests has a json method, its very useful for them.

Reply:
response = requests.get()
response.json
response.text 
is the pattern i was going for